lib/nimble_parsec/compiler.ex

```diff
@@ -983,20 +983,22 @@ defmodule NimbleParsec.Compiler do
     if ascii?(min) and ascii?(max) do
       <<" in the range ", ??, min, ?., ?., ??, max>>
     else
-      " in the range #{Integer.to_string(min)}..#{Integer.to_string(max)}"
+      " in the range #{none_ascii_to_string(min)}..#{none_ascii_to_string(max)}"
     end
   end
 
   defp inspect_bin_range(min) do
     if ascii?(min) do
       <<" equal to ", ??, min>>
     else
-      " equal to #{Integer.to_string(min)}"
+      " equal to #{none_ascii_to_string(min)}"
     end
   end
 
   defp ascii?(char), do: char >= 32 and char <= 126
 
+  defp none_ascii_to_string(none_ascii), do: inspect([none_ascii], binaries: :as_string)
```

Comment:
```suggestion
  defp none_ascii_to_string(none_ascii), do: inspect(<<none_ascii>>, binaries: :as_string)
```

Otherwise the binaries option has no effect. :)

Reply:
I always start tripping when I have to handle strings and binaries in combination with escaping.

But with `inspect(<<none_ascii>>, binaries: :as_string)` I got `"<<10>>"` for `?\n`.
I think `inspect([none_ascii])` will do the job and return `"'\\n'"`.

Reply:
It is because it should be `:as_strings` instead of `:as_string`. :)